Library/Homebrew/dev-cmd/sh.rb

```diff
@@ -44,16 +50,20 @@ def sh
     else
       "PS1=\"brew \\[\\033[1;32m\\]\\w\\[\\033[0m\\]$ \" #{ENV["SHELL"]}"
     end
-    ENV["VERBOSE"] = "1"
-    puts <<~EOS
-      Your shell has been configured to use Homebrew's build environment;
-      this should help you build stuff. Notably though, the system versions of
-      gem and pip will ignore our configuration and insist on using the
-      environment they were built under (mostly). Sadly, scons will also
-      ignore our configuration.
-      When done, type `exit`.
-    EOS
-    $stdout.flush
-    safe_system subshell
+    ENV["VERBOSE"] = "1" if args.verbose?
+    if cmd.present?
+      safe_system(ENV["SHELL"], "-c", cmd)
```

Comment:
This feels like it should also use `subshell` to be consistent.

Reply:
`subshell` is defined on lines 48-52 [above](https://github.com/Homebrew/brew/pull/8482/files#diff-6be2d74426eb397988e7ef728ed65931R48-R50) but otherwise it's the same `ENV["SHELL"]`. Calling `ENV["SHELL"]` (/bin/bash or /bin/zsh) via `safe_system` does launch a real subshell.

Reply:
How about putting the definition just in the `else` if it's only used there?

Reply:
sure!

Reply:
done.
I wonder, however, why `PS1=... /bin/bash` approach is used because `PS1='$' /bin/bash` doesn't work for me. Shall we be setting `ENV["PS1"]` instead?

Reply:
> I wonder, however, why `PS1=... /bin/bash` approach is used because `PS1='$' /bin/bash` doesn't work for me (I tweak and export PS1).

Can you elaborate on "doesn't work for me"?

Reply:
`PS1=... /bin/bash` assumes that no manipulations are made in `.bashrc` or `.bash_profile` files which are loaded automatically unless we add `--noprofile` and `--norc` to the `/bin/bash` call. I change in one of those files, so `PS1=... /bin/bash` doesn't work for me... and now, that I think about it, `ENV["PS1"] = ...` wouldn't fix it either...

Shall we add `--noprofile` and `--norc` ?

Reply:
I suspect this is primarily just to set a prompt if one wasn't already set rather than a specific one.

Reply:
Makes sense.

I wonder, though, about the `--noprofile` and `--norc` options: misconfiguration in any of those automatically loaded files will lead to problems with `brew sh`... So, unless we change `HOME`, adding these two flags might be useful.

Reply:
but, perhaps, not in this PR...

Reply:
Yes, not this PR. I'd pass on it in general because no-one has asked for it.